integration-tests/src/test/java/tachyon/worker/DataServerIntegrationTest.java

```diff
@@ -273,10 +298,7 @@ public void readThroughClientNonExistentTest() throws IOException, TachyonExcept
 
     RemoteBlockReader client =
         RemoteBlockReader.Factory.createRemoteBlockReader(mWorkerTachyonConf);
-    ByteBuffer result = client.readRemoteBlock(
-        new InetSocketAddress(block.getLocations().get(0).getWorkerAddress().getHost(),
-            block.getLocations().get(0).getWorkerAddress().getDataPort()),
-        maxBlockId + 1, 0, length);
```

Comment:
Before we were using `maxBlockId + 1` for the block id, but this changes it to `block.getBlockId()` in `readRemotely`. Is this intentional? How does this test pass?